src/fpylll/fplll/svpcvp.pyx

```diff
@@ -24,41 +23,82 @@ from .fplll cimport shortest_vector_pruning
 from fpylll.io cimport assign_Z_NR_mpz, mpz_get_python
+from fpylll.io import SuppressStream
 from fpylll.util import ReductionError
 
 from .integer_matrix cimport IntegerMatrix
 
-def shortest_vector(IntegerMatrix B, method="proved", int flags=SVP_DEFAULT, pruning=None, run_lll=True, max_aux_sols=0):
-    """Return a shortest vector. The result is guaranteed if method = "proved".
+def shortest_vector(IntegerMatrix B, method="fast", int flags=SVP_DEFAULT, pruning=True, preprocess=True, max_aux_solutions=0):
+    """Return a shortest vector. The result is guaranteed if ``method`` is "proved".
 
-    :param IntegerMatrix B:
+    :param B: A lattice basis
     :param method: One of "fast" or "proved".
     :param int flags:
-    :param pruning:
-    :param run_lll:
-    :param max_aux_sols:
-    :returns:
-    :rtype:
+    :param pruning: If ``True`` pruning parameters are computed by this function.
+    :param preprocess: Blocksize used for preprocessing; if  ``True`` a block size is picked.
+    :param max_aux_solutions: maximum number of additional short-ish solutions to return
 
     """
 
+    d = B.nrows
+
+    if pruning is True and d <= 10:
+        pruning = None # HACK: pruning in small dimensions can go wrong.
+
+    if d > MAX_ENUM_DIM:
+        raise NotImplementedError("This build of FPLLL is configured with a maximum enumeration dimension of %d."%MAX_ENUM_DIM)
+
     if B._type != ZT_MPZ:
         raise NotImplementedError("Only integer matrices over GMP integers (mpz_t) are supported.")
 
     cdef SVPMethod method_
+
     if method == "proved":
         method_ = SVPM_PROVED
+        if pruning is True:
+            pruning = None
+        if pruning is not None:
+            raise ValueError("Method 'proved' is incompatible with providing pruning parameters.")
     elif method == "fast":
         method_ = SVPM_FAST
     else:
         raise ValueError("Method '{}' unknown".format(method))
 
     cdef int r = 0
-
-    if run_lll:
-        lll_reduction(B)
+    s = load_strategies_json(BKZ.DEFAULT_STRATEGY)[-1]
+
+    if preprocess is True and d > s.block_size:
+        preprocess = max(min(d-10, s.block_size), 2)
+
+    if preprocess == 2: # just run LLL
+        B = LLL.reduction(B)
+    elif preprocess is True: # automatic choice
+        bkz_obj = BKZReduction(B)
+        bkz_obj.svp_reduction(0, d, BKZ.EasyParam(d))
+    elif preprocess and preprocess > 2: # make something work
+        preprocess = max(min(d-10, preprocess), 2)
+        bkz_obj = BKZReduction(B)
+        bkz_obj(BKZ.EasyParam(preprocess))
+
+    if pruning is True:
+        M = GSO.Mat(B)
+        M.update_gso()
+        for cost in (10, 20, 30, 40, 50):
+            try:
+                with SuppressStream():
+                    pruning = Pruning.run(M.get_r(0, 0), 2**cost, M.r(), 0.99, flags=Pruning.SINGLE|Pruning.GRADIENT)
+                pruning = pruning.coefficients
+                break
+            except RuntimeError:
+                pass
```

Comment:
What if they all fail ? I think this triggers an error down the road. How about:
- throwing a warning and setting pruning to no pruning (i.e. [1,1,1,1 ... ] ?

Reply:
Good idea, done.